packages/kit/src/runtime/server/respond.js

```diff
@@ -297,6 +297,10 @@ export async function respond(request, options, manifest, state) {
 			}
 		}
 
+		if (!event.platform && DEV && state.emulator?.platform) {
+			event.platform = await state.emulator.platform({ config: {}, prerender: false });
+		}
```

Comment:
@benmccann is this what you meant in https://github.com/sveltejs/kit/issues/11996#issuecomment-2211792494 ? I suppose one more check we need to add is if we're currently rendering a fallback